modify GameRenderer

nh0317 opened the pull request:
GameRenderer의 factor 수정했습니다.

또한 player와 ammo의 위치, 이동 속도를 약간 수정했습니다.

nh0317 committed:
modify GameRenderer

GameRenderer의 factor 수정했습니다.

또한 player와 ammo의 위치, 이동 속도를 약간 수정했습니다.

onebone commented:
확인했습니다.

한 가지 걸리는 게 있는데, GameRenderer에서 월드의 크기에 비례해서가 아니라 화면의 크기에 맞추기 위해서 factorWidth와 factorHeight를 따로 하는 것은 나쁘지 않은 것 같습니다. 다만 엔티티의 크기만 factorWidth과 factorHeight 중 하나만으로 크기를 조정하는데 이렇게 되면 실제 판정과 사용자의 눈에 보이는 피격 판정과 달라질 우려가 있을 것 같습니다.

제가 생각할 수 있는 해결 방법으로는

1. 폼의 크기만 정사각형으로 조정하고 나머지는 원래대로 되돌리기
2. 엔티티도 마찬가지로 동일 비율로 늘어나게 하기

입니다. 나머지는 좋은 것 같습니다!

nh0317 committed:
modify EntityToRect

entity의 비율도 동일하게 증가하도록 수정했습니다

nh0317 commented:
제가 해당사항을 미처 고려하지 못했습니다. 
2번째 방향으로 수정하여 커밋했습니다

nh0317 committed:
UI update

게임의 전반적인 ui를 구성했습니다.
게임의 컨셉을 바꾸는 것은 무리라고 생각하여 일단 기존의 space 컨셉을 유지했습니다.

또한 login과 회원 가입, ranking 출력 기능을 구현했습니다.

더 나은 아이디어 혹은 수정 사항이 있다면 언제든지 말씀해주세요

nh0317 committed:
Font update

깜빡하고 사용했던 폰트를 올리지 않아 추가로 올립니다.

혹시 폰트가 깨지면 알려주세요. 해결 방법을 찾아보거나 기본 폰트로 다시 작업하도록 하겠습니다.

onebone commented on Galaga/SignUp.cs:
```diff
@@ -0,0 +1,52 @@
+            FileAccess.FileAccess fileAccess = new FileAccess.FileAccess();
+            List<string[]> userInfo = fileAccess.readInfo();
+
+            if(txtID.Text is null || txtPW.Text is null)
```
제 환경에서는 TextBox가 비어 있어도 null이 되지는 않는데 한 번 확인해주실 수 있을까요?

onebone commented on Galaga/FileAccess/FileAccess.cs:
```diff
@@ -0,0 +1,48 @@
+{
+    class FileAccess
+    {
+        String path = System.IO.Directory.GetCurrentDirectory()+"./../../PlayerInfo/PlayerInfo.csv";
```
파일의 위치를 지정할 때는 이런 식으로 문자열을 직접 더하는 방식보다는

```c#
Path.GetFullPath(Path.Combine(Directory.GetCurrentDirectory(), "..", "..", "PlayerInfo", "PlayerInfo.csv"))
```

와 같은 코드로 각 환경에 맞는 경로 표시법을 사용하도록 하는 게 좋을 것 같습니다! 이 프로그램이 다른 환경에서 돌아갈 일이 없기야 하겠지만 바꿔서 나쁠 건 없을 것 같네요.

onebone commented on Galaga/FileAccess/FileAccess.cs:
```diff
@@ -0,0 +1,48 @@
+    class FileAccess
+    {
+        String path = System.IO.Directory.GetCurrentDirectory()+"./../../PlayerInfo/PlayerInfo.csv";
+        public List<string[]> readInfo()
```
이 함수는 사용할 멤버 변수가 없기 때문에 static 함수로 사용하는 게 나을 것 같습니다! path 멤버 변수도 마찬가지로 static으로 변경하면 좋을 것 같습니다.

onebone replied:
List<string[]>을 반환하는 대신 데이터를 담는 클래스를 따로 만들어서 반환하는 게 가독성이 더 좋을 것 같습니다


```c#
public List<Account> ReadInfo() {
    ....
}

public class Account {
    public string Id;
    public string Password;
    ....
}
```

onebone commented on Galaga/FileAccess/FileAccess.cs:
```diff
@@ -0,0 +1,48 @@
+            return csvList;
+        }
+
+        public void WriteInfo(String ID, String PW)
```
여기도 마찬가지로 static으로 바꾸는 게 좋을 것 같습니다

onebone commented on Galaga/Login.cs:
```diff
@@ -0,0 +1,57 @@
+            {
+                check = false;
+
+                for (int row = 0; row < PlayerInfoList.Count; row++)
```
이 부분은 굳이 이렇게 두 번씩 루프를 돌지 말고 위 루프문에서 일치하는 ID를 찾았을 때 비밀번호도 같이 비교하는 게 낫지 않을까요?

onebone commented on Galaga/PlayerInfo/PlayerInfo.csv:
```diff
@@ -0,0 +1,6 @@
+Lee,3,90,1234
```
계정 정보를 모아두는 파일인 것 같은데 git 레포에 올려서 좋을 건 없기 때문에 .gitignore에 추가하는 게 좋을 것 같습니다!

onebone commented on Galaga/UserPage.cs:
```diff
@@ -0,0 +1,63 @@
+            int userIndex=0;
+
+
+            for (int row = 0; row < userInfo.Count; row++)
```
여기서는 이렇게 반복문을 사용해서 찾는 것보다

```c#
var userIndex = userInfo.FindIndex(it => string.Equals(it[0], userName));
```

와 같은 함수형의 형태로 찾는 게 읽기 편할 것 같습니다!

onebone commented on Galaga/SignUp.cs:
```diff
@@ -0,0 +1,52 @@
+
+namespace Galaga
+{
+    public partial class sign_up : Form
```
C#의 권장 표기법에 따라 class SignUp으로 변경하는 게 좋을 것 같습니다!

nh0317 committed:
Modify Ui

말씀해주신 수정 사항 반영하여 수정했습니다.

또한, 외부 폰트를 윈폼에 적용하는 방법을 찾아서 적용해봤는데 폼을 여러 번 로드하는 경우 에러가 발생하여 기본 폰트('맑은 고딕')으로 폰트를 통일했습니다.
(UserPage에서 Ranking을 여러 번 열고 닫은 다음 start 버튼 클릭시 에러 발생)

nh0317 committed:
Update UserPage.cs

unserIndex 수정

onebone commented on Galaga/Login.cs:
```diff
@@ -0,0 +1,59 @@
+                    else
+                        MessageBox.Show("Check PW");
+                }
+                    check = true;
```
이 `check = true` 문은 무조건 실행되기 때문에 ID가 없어도 Check ID 메시지박스가 표시되지 않습니다

onebone commented:
`Galaga/PlayerInfo/PlayerInfo.csv` 파일이 .gitignore에는 추가됐지만 커밋에서 제거되지 않았습니다.

```bash
git rm --cached Galaga/PlayerInfo/PlayerInfo.csv
```

명령어를 사용하시면 깃에서 완전히 제거하실 수 있으니 참고부탁드립니다!

nh0317 committed:
Delete PlayerInfo.csv

캐시 삭제했습니다.

nh0317 committed:
Update Login.cs

login 관련 부분 수정했습니다.
